### ISSUE TITLE
OpenAPI 3.0+ null Type Parsing Not Supported

### ISSUE THREAD

**Issue by LeePui** [langgenius/dify#2605]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Dify version

0.5.6

### Cloud or Self Hosted

Self Hosted (Docker)

## Description
dify does not currently support parsing OpenAPI 3.0+ specifications that use the `null` type as part of a `anyOf`/`oneOf` construct. According to the OpenAPI 3.0+ specification, fields can be specified as nullable either by using the `nullable` keyword or by including `null` as one of the types in an `anyOf`/`oneOf` construct.

### Steps to Reproduce
1. Include a field in an OpenAPI 3.0+ specification with a `null` type as part of an `anyOf` construct.
2. Attempt to parse the specification with dify.
3. Observe that the `null` type is not handled correctly.

### 3.0+ openapi format:
```
"properties": {
          "chatId": {
            "anyOf": [
              {
                "type": "string"
              },
              {
                "type": "null"
              }
            ],
            "title": "Chatid",
            "description": "chatId"
          },
```

### ✔️ Expected Behavior

When presented with a schema that includes a field with types including `null` using the `anyOf` construct, dify should correctly interpret this as a field that can have a `null` value or the other specified type.

### ❌ Actual Behavior

![image](https://github.com/langgenius/dify/assets/14257862/5bf32ff2-d73f-4d88-84a0-d3819b1da3a8)

**Comment by Yeuoly** [comment]
Could you pls provider the openapi schema you are using for this?

**Comment by LeePui** [comment]
> Could you pls provider the openapi schema you are using for this?

- fastapi==0.104.0

sure，I provide an interface using Python and FastAPI. Here is my base model：
```python
class DrawBaseModel(BaseModel):
    """a test model
    """
    chatId: Optional[str] = Field(None, description="chatId")
    prompt: str = Field(..., description="prompt")
```

Here is openapi.json like:
```
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "servers": [
    {
      "url": "https://xxx.com"
    }
  ],
  "paths": {
    "/draw": {
      "post": {
        "summary": "draw pic",
        "description": "Draw a picture according to the prompts",
        "x-openai-isConsequential": false,
        "operationId": "draw_post",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/DrawBaseModel"
              }
            }
          },
          "required": true
        }
      }
    }
  },
  "components": {
    "schemas": {
      "DrawBaseModel": {
        "properties": {
          "chatId": {
            "anyOf": [
              {
                "type": "string"
              },
              {
                "type": "null"
              }
            ],
            "title": "Chatid",
            "description": "chatId"
          },
          "prompt": {
            "type": "string",
            "title": "Prompt",
            "description": "prompt"
          }
        },
        "type": "object",
        "title": "test_model",
        "description": "a test model"
      }
    }
  }
}
```

When I use Dify to create custom tools and then test, I get the following error:
![image](https://github.com/langgenius/dify/assets/14257862/16e6f615-f091-4986-9295-c2280ca130ed)

`/console/api/workspaces/current/tool-provider/api/test/pre`  response:
![image](https://github.com/langgenius/dify/assets/14257862/77e92d04-90f4-4a73-ad79-dc85c6928138)

If the above `openapi.json` is imported into GPTS, it can be parsed correctly.

If I change the 'chatId' inside the above JSON to this, Dify will be able to parse it correctly:
```
"chatId": {
    "type": "string",
    "title": "Chatid",
    "description": "chatId"
}
```

**Comment by Yeuoly** [comment]
Thanks for your supports, it has been fixed and merged into main branch.

**Comment by crazywoola** [comment]
Fixed #2625
